dev/tasks/python-wheels/osx-build.sh

```diff
@@ -119,37 +119,37 @@ function build_wheel {
     pushd cpp
     mkdir build
     pushd build
-    cmake -DCMAKE_BUILD_TYPE=Release \
-          -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
-          -DARROW_VERBOSE_THIRDPARTY_BUILD=ON \
-          -DARROW_BUILD_TESTS=OFF \
+    cmake -DARROW_BOOST_USE_SHARED=ON \
           -DARROW_BUILD_SHARED=ON \
-          -DARROW_BOOST_USE_SHARED=ON \
+          -DARROW_BUILD_TESTS=OFF \
+          -DARROW_FLIGHT=ON \
+          -DARROW_GANDIVA=${BUILD_ARROW_GANDIVA} \
           -DARROW_JEMALLOC=ON \
+          -DARROW_ORC=OFF \
+          -DARROW_PARQUET=ON \
           -DARROW_PLASMA=ON \
-          -DARROW_RPATH_ORIGIN=ON \
+          -DARROW_PROTOBUF_USE_SHARED=OFF \
           -DARROW_PYTHON=ON \
+          -DARROW_RPATH_ORIGIN=ON \
+          -DARROW_VERBOSE_THIRDPARTY_BUILD=ON \
+          -DARROW_WITH_BROTLI=ON \
           -DARROW_WITH_BZ2=ON \
-          -DARROW_WITH_ZLIB=ON \
-          -DARROW_WITH_ZSTD=ON \
           -DARROW_WITH_LZ4=ON \
           -DARROW_WITH_SNAPPY=ON \
-          -DARROW_WITH_BROTLI=ON \
-          -DARROW_PARQUET=ON \
-          -DARROW_GANDIVA=${BUILD_ARROW_GANDIVA} \
-          -DARROW_ORC=OFF \
+          -DARROW_WITH_ZLIB=ON \
+          -DARROW_WITH_ZSTD=ON \
           -DBOOST_ROOT="$arrow_boost_dist" \
           -DBoost_NAMESPACE=arrow_boost \
-          -DARROW_FLIGHT=ON \
-          -DgRPC_SOURCE=SYSTEM \
+          -DBoost_NO_BOOST_CMAKE=ON \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
+          -DMAKE=make \
+          -DOPENSSL_USE_STATIC_LIBS=ON  \
           -Dc-ares_SOURCE=BUNDLED \
+          -DgRPC_SOURCE=SYSTEM \
           -Dzlib_SOURCE=BUNDLED \
-          -DARROW_PROTOBUF_USE_SHARED=OFF \
-          -DOPENSSL_USE_STATIC_LIBS=ON  \
-          -DOPENSSL_ROOT_DIR=$(brew --prefix openssl@1.1) \
-          -DMAKE=make \
```

Comment:
Sorted and added `-DBoost_NO_BOOST_CMAKE=ON` to not use Boost installed by Homebrew.